### PR TITLE
Powershell-aware panic environment variable suggestions in backtraces and grammatical consistency.

### LINKED ISSUE #98379: Suggest powershell syntax for setting RUST_BACKTRACE in the panic result on Windows
Currently on Windows when a program panics and the default panic handler is run one of the lines it prints is: ``note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace``.

The problem is this **is not** the syntax that powershell (which I think is the default and largest shell used on Windows) uses to set environment variables. 

I think ``note: `RUST_BACKTRACE=1` (or PowerShell `$env:RUST_BACKTRACE=1`) environment variable to display a backtrace`` would be more correct and better for beginners.

Should be pretty simple change adding a different `cfg!` branch for following code https://github.com/rust-lang/rust/blob/352e621368c31d7b4a6362e081586cdb931ba020/library/std/src/sys_common/backtrace.rs#L104-L110

### PR DESCRIPTION
**Pull request by ghost**
Fixes #98379 

This PR also extends the original issue by being cross-platform, rather than just Windows (powershell can run on Win, Mac, Linux and friends!)

# Screenshot
image

### CODE REVIEW

File: library/std/src/panicking.rs [hunk 1]
```diff
@@ -296,10 +296,17 @@ fn default_hook(info: &PanicInfo<'_>) {
...
-                    );
+                    let _ = if let Ok(_) = crate::env::var("PSModulePath") {
+                        writeln!(
+                            err,
+                            "note: run with `$env:RUST_BACKTRACE=1` environment variable to display a backtrace."
+                        )
+                    } else {
+                        writeln!(
+                            err,
+                            "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace"
```
Comment (killerswin2) [inline library/std/src/panicking.rs:307]: `"note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace."`

Comment (killerswin2) [reply library/std/src/panicking.rs:307]: oof, add a period the end.

Comment (gimbles) [reply library/std/src/panicking.rs:307]: https://github.com/rust-lang/rust/blob/master/library/std/src/panicking.rs#L301

The original msg doesn't seem to include a period at the end, so I think I should *remove* them instead :)

Comment (gimbles) [reply library/std/src/panicking.rs:307]: Hm, that's odd, the RUST_BACKTRACE=full msg does include a period.

Meh doesn't really matter I'll add it.

File: library/std/src/panicking.rs [hunk 2]
```diff
@@ -296,10 +296,17 @@ fn default_hook(info: &PanicInfo<'_>) {
             }
             Some(BacktraceStyle::Off) => {
                 if FIRST_PANIC.swap(false, Ordering::SeqCst) {
-                    let _ = writeln!(
-                        err,
-                        "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace"
-                    );
+                    let _ = if let Ok(_) = crate::env::var("PSModulePath") {
```
Comment (Milo123459) [inline library/std/src/panicking.rs:299]: Couldn't you just check the $SHELL env variable?

Comment (gimbles) [reply library/std/src/panicking.rs:299]: Does not exist on Powershell by default, just confirmed on Linux and Windows.

**Pull request closed without merging**